src/software/sensor_fusion/sensor_fusion_test.cpp

```diff
@@ -1010,3 +1010,289 @@ TEST_F(SensorFusionTest, breakbeam_fail_test_ssl)
+
+    std::vector<Robot> injured_robots =
+        sensor_fusion.getWorld().value().friendlyTeam().getInjuredRobots();
+    EXPECT_EQ(injured_robots.size(), 1);
+
+    if (injured_robots.size() == 1)
+    {
+        EXPECT_EQ(injured_robots[0].id(), 1);
+    }
+}
+
+TEST_F(SensorFusionTest, test_detect_front_right_motor_fault)
+{
+    std::vector<TbotsProto::RobotStatus> robot_status_msgs;
+    auto robot_msg = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg->set_robot_id(1);
+    robot_msg->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg->mutable_motor_status()->mutable_front_right()->add_motor_faults(
+        TbotsProto::MotorFault::DRIVER_OVERTEMPERATURE);
+    robot_status_msgs.push_back(*robot_msg);
+
+    auto robot_msg2 = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg2->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg2->set_robot_id(2);
+    robot_status_msgs.push_back(*robot_msg2);
+
+    SensorProto sensor_msg;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    sensor_fusion.detectInjuredRobots(robot_status_msgs);
+
+    std::vector<Robot> injured_robots =
+        sensor_fusion.getWorld().value().friendlyTeam().getInjuredRobots();
+    EXPECT_EQ(injured_robots.size(), 1);
+
+    if (injured_robots.size() == 1)
+    {
+        EXPECT_EQ(injured_robots[0].id(), 1);
+    }
+}
+
+TEST_F(SensorFusionTest, test_detect_front_left_motor_fault)
+{
+    std::vector<TbotsProto::RobotStatus> robot_status_msgs;
+    auto robot_msg = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg->set_robot_id(1);
+    robot_msg->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg->mutable_motor_status()->mutable_front_left()->add_motor_faults(
+        TbotsProto::MotorFault::DRIVER_OVERTEMPERATURE_PREWARNING);
+    robot_status_msgs.push_back(*robot_msg);
+
+    auto robot_msg2 = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg2->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg2->set_robot_id(2);
+    robot_status_msgs.push_back(*robot_msg2);
+
+    SensorProto sensor_msg;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    sensor_fusion.detectInjuredRobots(robot_status_msgs);
+
+    std::vector<Robot> injured_robots =
+        sensor_fusion.getWorld().value().friendlyTeam().getInjuredRobots();
+    EXPECT_EQ(injured_robots.size(), 1);
+
+    if (injured_robots.size() == 1)
+    {
+        EXPECT_EQ(injured_robots[0].id(), 1);
+    }
+}
+
+TEST_F(SensorFusionTest, test_detect_back_right_motor_fault)
```

Comment:
to avoid duplicating all the tests, you can run a `TEST_P` and parameterize these motor_fault tests. here's an example: https://github.com/UBC-Thunderbots/Software/blob/1ee0eadd804f4f3bc541a0965702cd21e397caf6/src/shared/uart_framing/uart_framing_test.cpp#L55